tests/lib/fdtable/src/main.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <zephyr.h>
+#include <sys/fdtable.h>
+
+void test_z_reserve_fd(void)
+{
+
+    int fd = -1;
+    
+    fd = z_reserve_fd(); //function being tested
+
+    zassert_equal(fd >= 0, true, "fd not changed");
+
```

Comment:
probably need to free the fd after reserving it with z_free_fd(), there are only a total of CONFIG_POSIX_MAX_FDs available